test/e2e/network/network_policy.go

```diff
@@ -532,12 +640,601 @@ var _ = SIGDescribe("NetworkPolicy", func() {
+			err = e2epod.WaitForPodRunningInNamespace(f.ClientSet, nsBpodServerA)
+			framework.ExpectNoError(err, "Error occurred while waiting for pod status in namespace: Running.")
+
+			err = e2epod.WaitForPodRunningInNamespace(f.ClientSet, nsBpodServerB)
+			framework.ExpectNoError(err, "Error occurred while waiting for pod status in namespace: Running.")
+
+			defer cleanupServerPodAndService(f, nsBpodServerA, nsBserviceA)
+			defer cleanupServerPodAndService(f, nsBpodServerB, nsBserviceB)
+
+			ginkgo.By("Creating a network policy for the server which allows traffic only to a server in different namespace.")
+			protocolUDP := v1.ProtocolUDP
+			policyAllowToServerInNSB := &networkingv1.NetworkPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: nsA.Name,
+					Name:      "allow-to-ns-b-server-a-via-namespace-selector",
+				},
+				Spec: networkingv1.NetworkPolicySpec{
+					// Apply this policy to the client
+					PodSelector: metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"pod-name": "client-a",
+						},
+					},
+					PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeEgress},
+					// Allow traffic only to server-a in namespace-b
+					Egress: []networkingv1.NetworkPolicyEgressRule{
+						{
+							Ports: []networkingv1.NetworkPolicyPort{
+								// Allow DNS look-ups
+								{
+									Protocol: &protocolUDP,
+									Port:     &intstr.IntOrString{Type: intstr.Int, IntVal: 53},
+								},
+							},
+						},
+						{
+							To: []networkingv1.NetworkPolicyPeer{
+								{
+									NamespaceSelector: &metav1.LabelSelector{
+										MatchLabels: map[string]string{
+											"ns-name": nsBName,
+										},
+									},
+									PodSelector: &metav1.LabelSelector{
+										MatchLabels: map[string]string{
+											"pod-name": nsBpodServerA.ObjectMeta.Labels["pod-name"],
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+
+			policyAllowToServerInNSB, err = f.ClientSet.NetworkingV1().NetworkPolicies(f.Namespace.Name).Create(policyAllowToServerInNSB)
+			framework.ExpectNoError(err, "Error occurred while creating policy: policyAllowToServerInNSB.")
+			defer cleanupNetworkPolicy(f, policyAllowToServerInNSB)
+
+			ginkgo.By("Creating client-a, in 'namespace-a', which should be able to contact the server-a in namespace-b.", func() {
+				testCanConnect(f, nsA, "client-a", nsBserviceA, 80)
+			})
+			ginkgo.By("Creating client-a, in 'namespace-a', which should not be able to contact the server-b in namespace-b.", func() {
+				testCannotConnect(f, nsA, "client-a", nsBserviceB, 80)
+			})
+			ginkgo.By("Creating client-a, in 'namespace-a', which should not be able to contact the server in namespace-a.", func() {
+				testCannotConnect(f, nsA, "client-a", service, 80)
+			})
+		})
+
+		ginkgo.It("should enforce multiple ingress policies with ingress allow-all policy taking precedence based on PodSelector [Feature:NetworkPolicy]", func() {
```

Comment:
"based on PodSelector" doesn't make sense as part of the test description here since the second policy uses an empty pod selector. (Likewise for the egress test that follows.)

Reply:
Updated the comments.